Unset need_charset_alias when building for musl

localcharset uses ac_cv_gnu_library_2_1 from glibc21.m4
which actually shoudl be fixed in gnulib and then all downstream
projects will get it eventually. For now we apply the fix to
coreutils

Upstream-Status: Pending

Signed-off-by: Khem Raj <raj.khem@gmail.com>
Index: libidn2-2.0.5/unistring/Makefile.am
===================================================================
--- libidn2-2.0.5.orig/unistring/Makefile.am
+++ libidn2-2.0.5/unistring/Makefile.am
@@ -293,7 +293,7 @@ install-exec-localcharset: all-local
 	  case '$(host_os)' in \
 	    darwin[56]*) \
 	      need_charset_alias=true ;; \
-	    darwin* | cygwin* | mingw* | pw32* | cegcc*) \
+	    darwin* | cygwin* | mingw* | pw32* | cegcc* | linux-musl*) \
 	      need_charset_alias=false ;; \
 	    *) \
 	      need_charset_alias=true ;; \
